lib/matrix_api_lite/generated/api.dart

```diff
@@ -1993,12 +1996,14 @@ class Api {
   /// The joined room ID.
   Future<String> joinRoom(String roomIdOrAlias,
       {List<String>? serverName,
+      List<String>? via,
       String? reason,
       ThirdPartySigned? thirdPartySigned}) async {
     final requestUri = Uri(
         path: '_matrix/client/v3/join/${Uri.encodeComponent(roomIdOrAlias)}',
         queryParameters: {
           if (serverName != null) 'server_name': serverName,
+          if (via != null) 'via': via,
```

Comment:
You should really just set both when passing one. serverName is deprecated, but should still be sent for compat. I guess you could add a compat shim in the client for it.